src/librustdoc/html/render/cache.rs

```diff
@@ -373,18 +373,15 @@ crate fn get_all_types<'tcx>(
     tcx: TyCtxt<'tcx>,
 ) -> (Vec<TypeWithKind>, Vec<TypeWithKind>) {
     let mut all_types = Vec::new();
+    let mut args = Vec::new();
     for arg in decl.inputs.values.iter() {
         if arg.type_.is_self_type() {
             continue;
         }
-        // FIXME: performance wise, it'd be much better to move `args` declaration outside of the
-        // loop and replace this line with `args.clear()`.
-        let mut args = Vec::new();
+        args.clear();
```

Comment:
Is this clear necessary? Seems like it gets drained if it's not empty anyway, meaning it's impossible for it ever to not be empty here.

Reply:
This is a very good point, thanks!